scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/JsonDiff.java

```diff
@@ -251,81 +250,77 @@ private void computeArrayNodeDiffs(
 
       // Explicitly clear all attribute values.
       operations.add(PatchOperation.remove(path));
+      return;
     }
-    else
+
+    // Go through each value and try to individually patch them first
```

Comment:
The changes below this look messy, but it was an effort to remove multiple layers of nesting. Most lines are not changed, with the exception of switching to `instanceof`.